File: csharp/src/Training.Workshop/M5/S12BridgeMethods/Step2/StandardRule.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M5.S12BridgeMethods.Step2;

/// <summary>Krok 2: reguła implementuje tylko metodę dla swojego biletu - typ i "most" daje kontrakt.</summary>
public sealed class StandardRule : IPriceRule<StandardTicket>
{
    public Money Apply(StandardTicket ticket)
    {
        return ticket.BasePrice;
    }
}

File: typescript/src/workshop/m5/s12_bridgemethods/start/StandardRule.ts
import type { Money } from '../../../shared/Money.js';
import type { StandardTicket } from '../StandardTicket.js';

/** Start: reguła cenowa bez wspólnego typu - rejestr znajduje jej typ biletu po nazwie klasy. */
export class StandardRule {
  apply(ticket: StandardTicket): Money {
    return ticket.basePrice();
  }
}
